madsbangh/EasyButtons
# Resolve "Make EasyButtons a package manager package"

### Summary
Make repository Unity Package Manger compatible. Add GitHub action to create upm branch via CI.

### Details
Do not ignore .meta files as they are needed for UPM package to be valid. Rename examples folder to follow Unity package layout (see mention of samples in packages [here](https://forum.unity.com/threads/samples-in-packages-manual-setup.623080/)). Update README.md to include installation instructions when using UPM package.

### How to test
Try to add package to another Unity project via local install. Note that samples folder should be renamed into `Samples~` to be able to add samples via package manager UI.

To check GitHuib action correctness, please check branch created for test action:
* https://github.com/rfadeev/EasyButtons/tree/13-unity-package-manager-support-github-action-test

And resulting upm branch generated automatically:
* https://github.com/rfadeev/EasyButtons/tree/upm

Test CI run:
* https://github.com/rfadeev/EasyButtons/actions/runs/33668822

To install package via upm branch, update Unity `manifest.json` as follows (tested with Unity 2019.3.0f6):
```json
{
  "dependencies": {
    "com.madsbangh.easybuttons": "https://github.com/rfadeev/EasyButtons.git#upm"
  }
}
```

Closes #13
